json_feature.cpp

```diff
@@ -1,27 +1,38 @@
 
 #include "json_feature.hpp"
 
-void JSONFeature::add_properties(const osmium::OSMObject& object, const char* id_name) {
+void JSONFeature::add_properties(const osmium::OSMObject& object) {
     m_writer.String("properties");
 
     m_writer.StartObject();
 
-    m_writer.String(id_name);
-    m_writer.Int64(object.id());
+    m_writer.String((m_attr_prefix + "id").c_str());
+    m_writer.Int64(object.type() == osmium::item_type::area ? osmium::area_id_to_object_id(object.id()) : object.id());
+
+    m_writer.String((m_attr_prefix + "type").c_str());
+    if (object.type() == osmium::item_type::area) {
+        if ((object.positive_id() & 0x1) == 0) {
```

Comment:
Ok, that makes sense. But I still end up with the same error (with `const` sprinkled in). What I thought maybe it was getting at is that you can't pass an OSMObject into the Area constructor, leading to this part?

```
candidate constructor not viable: requires 0 arguments, but 1 was provided
```